docs/pipeline.rst

```diff
@@ -50,24 +53,39 @@ The resulting graph is:
 Current pipeline
 """"""""""""""""
 
-Subgraphs that do not contribute to the pipeline output are automatically pruned.
-If an operator has side effects (e.g. PythonFunction operator family), it cannot be invoked
+Sub-graphs that do not contribute to the pipeline output are automatically pruned.
+If an operator has side effects (e.g. ``PythonFunction`` operator family), it cannot be invoked
 without setting the current pipeline. Current pipeline is set implicitly when the graph is
-defined inside derived pipeline's `define_graph` method. Otherwise, it can be set using context
-manager (`with` statement)::
+defined inside derived pipelines' :meth:`Pipeline.define_graph` method.
+Otherwise, it can be set using context manager (``with`` statement)::
 
-    pipe = dali.pipeline.Pipeline(batch_size = N, num_threads = 3, device_id = 0)
+    pipe = dali.Pipeline(batch_size=N, num_threads=3, device_id=0)
     with pipe:
-        src = dali.ops.ExternalSource(my_source, num_outputs = 2)
+        src = dali.ops.ExternalSource(my_source, num_outputs=2)
         a, b = src()
         pipe.set_outputs(a, b)
 
+The code above may also be expressed using some Python syntactic-sugar::
```

Comment:
It's not just syntactic sugar (and that term elicits mixed feelings at best). Instead, we should focus on _which_ code is executed in current pipeline scope.
```suggestion
When a pipeline is constructed with a pipeline decorator, the entire decorated function is executed within the scope
of the newly created pipeline::
```